src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs

```diff
@@ -0,0 +1,164 @@
+               .SignatureBytes.Returns(new byte[64]);
+
+            var transactionValidator = new TransactionValidator(subbedLogger, subbedContext);
+            var privateKey = subbedContext.GeneratePrivateKey();
+
+            var validTransactionBroadcast = new TransactionBroadcast
+            {
+                PublicEntries =
+                {
+                    new PublicEntry
+                    {
+                        Base = new BaseEntry
+                        {
+                            SenderPublicKey = privateKey.GetPublicKey().Bytes.ToByteString()
+                        }
+                    }
+                }
+            };
+
+            var signature = new Signature
+            {
+                // sign an actual TransactionBroadcast object
+                RawBytes = subbedContext.Sign(privateKey, validTransactionBroadcast.ToByteArray(), Arg.Any<byte[]>())
+                   .SignatureBytes.ToByteString(),
+                SigningContext = new SigningContext
+                {
+                    NetworkType = NetworkType.Devnet,
+                    SignatureType = SignatureType.TransactionPublic
+                }
+            };
+
+            validTransactionBroadcast.Signature = signature;
+
+            var result = transactionValidator.ValidateTransaction(validTransactionBroadcast);
+            result.Should().BeTrue();
+        }
+        
+        [Fact]
+        public void TransactionValidator_ValidateTransactionSignature_returns_false_for_invalid_transaction_signature_verification()
+        {
+            var subbedLogger = Substitute.For<ILogger>();
+            var subbedContext = Substitute.For<ICryptoContext>();
+            
+            var privateKey = subbedContext.GeneratePrivateKey();
+
+            // raw un-signed tx message
+            var validTransactionBroadcast = new TransactionBroadcast
+            {
+                PublicEntries =
+                {
+                    new PublicEntry
+                    {
+                        Base = new BaseEntry
+                        {
+                            SenderPublicKey = privateKey.GetPublicKey().Bytes.ToByteString()
+                        }
+                    }
+                }
+            };
+
+            var txSig = new Signature
+            {
+                RawBytes = new byte[64].ToByteString(), //random bytes that are not of a signed TransactionBroadcast Object
+                SigningContext = new SigningContext
+                {
+                    NetworkType = NetworkType.Devnet,
+                    SignatureType = SignatureType.TransactionPublic
+                }
+            };
+            
+            subbedContext.GetSignatureFromBytes(Arg.Is(txSig.ToByteArray()), Arg.Is(privateKey.GetPublicKey().Bytes))
+               .ReturnsForAnyArgs(Substitute.For<ISignature>());
+            
+            subbedContext.Verify(Arg.Any<ISignature>(), // @TODO be more specific
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4386440)